File: Python/trap_forces_efficiencies.py
# These calculations are based on Ashkin's article "Forces of a single-beam
# gradient laser trap on a dielectric sphere in the ray optics regime

import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from functions import *


sns.set()

t = np.linspace(0, np.pi / 2, 1000)
t_deg = t * 180 / np.pi
pol = np.pi / 4

plt.figure(figsize=(13, 8))
plt.plot(t_deg, q_s(t, pol), '--', label='Q_s')
plt.plot(t_deg, -q_g(t, pol), '-.', label='Q_g')
plt.plot(t_deg, q_mag(t, pol), label='Q_t')
plt.grid()
plt.xlabel(r'$\theta$, deg', fontsize=18)
plt.ylabel('Q', fontsize=18)
plt.legend(fontsize=18)
plt.title('Beam efficiencies', fontsize=20)
plt.show()

File: Matlab/trap_forces_efficiencies.m
close all
clear
format compact
clc

% These calculations are based on Ashkin's article "Forces of a single-beam
% gradient laser trap on a dielectric sphere in the ray optics regime

load_constants

t = linspace(0, pi/2, 1000);
t_deg = t * 180/pi;
psi = pi / 4;

figure
plot(t_deg, qs_factor(t, psi, n1, n2), ...
    t_deg, -qg_factor(t, psi, n1, n2), ...
    t_deg, qmag_factor(t, psi, n1, n2));
grid
title(['\psi = ', num2str(psi * 180/pi), '^\circ'])
xlabel('\theta, ^\circ')
ylabel('Q')
legend('Q_s','Q_g','Q_t','location','northwest')